### PR TITLE
@ #26 | should let the users choose the api version

### LINKED ISSUE #26: should let the users choose the api version of their deployment
see:
- https://github.com/helm/charts/pull/20185
- https://github.com/kubernetes/kubernetes/issues/55894

I got error when using the duplicity helm chart with the latest teracy-dev-k8s:

```
$ helm upgrade --install inspector -f join-prod.yaml teracy-incubator/pv-join --version=0.1.0-SNAPSHOT
Release "inspector" does not exist. Installing it now.
Error: unable to build kubernetes objects from release manifest: unable to recognize "": no matches for kind "Deployment" in version "apps/v1beta2"
```

Please fix for all the current helm charts

### PR DESCRIPTION
**Pull request by hieptranquoc**
connect #26 
please check this PR! Thank you.

### CODE REVIEW

**Comment by hieptranquoc** [comment]
@hoatle please check this PR!

**Comment by hoatle** [comment]
@hieptranquoc please follow the convention from https://github.com/helm/charts/pull/20185/files
with:

```yaml
# Metadata surrounding the kubernetes distribution being deployed to.
kubeMeta:
  # The api that Deployments are under.
  deploymentApiVersion: apps/v1
  batchApiVersion: batch/v1
 ## add more types here
```

**Review by hoatle — CHANGES_REQUESTED** [review]
@hieptranquoc please follow the convention and fix the travis-ci build

File: .travis.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ install:
 - wget -q ${HELM_URL}/${HELM_TARBALL}
 - tar xzfv ${HELM_TARBALL}
 - PATH=`pwd`/linux-amd64/:$PATH
-- helm init --client-only
+- helm init
```
Comment (hoatle) [inline .travis.yml]: remove this

File: incubator/duplicity/templates/job.yaml [hunk 2]
```diff
@@ -13,7 +13,7 @@ metadata:
     # job is considered part of the release.
     "helm.sh/hook": post-install,post-upgrade
     "helm.sh/hook-weight": "-5"
-    "helm.sh/hook-delete-policy": before-hook-creation, hook-succeeded
```
Comment (hoatle) [inline incubator/duplicity/templates/job.yaml]: why removing `before-hook-creation`?

**Comment by hieptranquoc** [comment]
I applied for your comment! Please check this PR again

File: incubator/mysqlha/templates/statefulset.yaml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-apiVersion: apps/v1beta1
+apiVersion: {{ default "apps/v1beta1" .Values.kubeMeta.statefulsettApiVersion }}
```
Comment (hoatle) [inline incubator/mysqlha/templates/statefulset.yaml]: let's use `.Values.kubeMeta.appsApiVersion` and  default `apps/v1`

File: incubator/pv-join/templates/deployment.yaml [hunk 4]
```diff
@@ -1,4 +1,4 @@
-apiVersion: apps/v1beta2
+apiVersion: {{ default "apps/v1beta2" .Values.kubeMeta.deploymentApiVersion }}
```
Comment (hoatle) [inline incubator/pv-join/templates/deployment.yaml]: let's use `.Values.kubeMeta.appsApiVersion` and  default `apps/v1`

File: incubator/pv-join/values.yaml [hunk 5]
```diff
@@ -2,6 +2,9 @@
 # This is a YAML-formatted file.
 # Declare variables to be passed into your templates.
 
+kubeMeta:
+  deploymentApiVersion: apps/v1beta2
```
Comment (hoatle) [inline incubator/pv-join/values.yaml]: `appsApiVersion: apps/v1`

File: incubator/mysqlha/values.yaml [hunk 6]
```diff
@@ -1,6 +1,9 @@
 ## mysql image version
 ## ref: https://hub.docker.com/r/library/mysql/tags/
 ##
+kubeMeta:
+  statefulsettApiVersion: apps/v1beta1
```
Comment (hoatle) [inline incubator/mysqlha/values.yaml]: `appsApiVersion: apps/v1`

**Review by hoatle — CHANGES_REQUESTED** [review]
@hieptranquoc please see my comments and `$ kubectl api-versions` to see all available versions to set the default values.

**Comment by hieptranquoc** [comment]
please check this PR again! Thank you.

**Comment by hoatle** [comment]
see all the available api versions of a cluster:

```
$ kubectl api-versions
acme.cert-manager.io/v1alpha2
admissionregistration.k8s.io/v1beta1
apiextensions.k8s.io/v1beta1
apiregistration.k8s.io/v1
apiregistration.k8s.io/v1beta1
apps/v1
apps/v1beta1
apps/v1beta2
authentication.k8s.io/v1
authentication.k8s.io/v1beta1
authorization.k8s.io/v1
authorization.k8s.io/v1beta1
autoscaling/v1
autoscaling/v2beta1
autoscaling/v2beta2
batch/v1
batch/v1beta1
cert-manager.io/v1alpha2
certificates.k8s.io/v1beta1
coordination.k8s.io/v1
coordination.k8s.io/v1beta1
extensions/v1beta1
internal.autoscaling.k8s.io/v1alpha1
metrics.k8s.io/v1beta1
migration.k8s.io/v1alpha1
networking.gke.io/v1beta1
networking.k8s.io/v1
networking.k8s.io/v1beta1
nfs.rook.io/v1alpha1
node.k8s.io/v1beta1
nodemanagement.gke.io/v1alpha1
policy/v1beta1
rbac.authorization.k8s.io/v1
rbac.authorization.k8s.io/v1beta1
scalingpolicy.kope.io/v1alpha1
scheduling.k8s.io/v1
scheduling.k8s.io/v1beta1
storage.k8s.io/v1
storage.k8s.io/v1beta1
```

and we can get convention: `<normalizedType(apiName)>ApiVersion: <default_value>`
the default value is the latest stable recent version

for example:
`acme.cert-manager.io/v1alpha2` -> `acmeCertManagerIOApiVersion: v1alpha2`
`apps/v1` -> `appsApiVersion: v1`
`storage.k8s.io/v1` -> `storageK8sIOApiVersion: v1`
and the list goes on...

**Comment by hoatle** [comment]
seems like we'll deprecate and not maintain the `zalenium` helm chart anymore, so just ignore it.

**Review by hoatle — APPROVED** [review]
LGTM

**Pull request merged**